ably/realtime/connection.py

```diff
@@ -142,6 +166,13 @@ async def ws_read_loop(self):
                 self.__websocket = None
                 self.__closed_future.set_result(None)
                 break
+            if action == ProtocolMessageAction.HEARTBEAT:
+                if self.__ping_future:
+                    # Resolve on heartbeat from ping request.
+                    # TODO: Handle Normal heartbeat if required
+                    if self.__ping_id == msg["id"]:
```

Comment:
I think if a regular heartbeat is received this will throw a `KeyError`. You should use `msg.get("id")` instead